core/includes/icon.inc

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * @file
+ * Provides the Backdrop API for placing icons.
+ */
+
+/**
+ * Return the markup for outputting an icon by a machine name.
+ *
+ * This will first check if an icon is provided by the active theme (in the
+ * theme's "icons" directory). If so, that icon will be used. Otherwise the
+ * icon will be pulled from whichever module provided the icon through
+ * hook_icon_info().
+ *
+ * @param string $icon_name
+ *   The icon name such as "home", "info", etc. Do not include the file
+ *   extension.
+ * @param array $options
+ *   An array of options to pass to the icon. Keys include:
+ *   - alt: An alternative text to be used for screen readers. If not specified
+ *     the icon will be marked as decorative with aria-hidden="true". Note that
+ *     in the case of SVG icons, this will rendered as a <title> attribute
+ *     within the <svg> tag.
+ *   - attributes: Any additional attributes to add to the icon, including
+ *     classes (as an array), name, id, etc.
+ *   - wrapper_attributes: Same as attributes but applied to the span wrapper
+ *     around the icon.
+ *   - immutable: Boolean value if this icon is not allowed to be overridden
+ *     by themes or modules. If an icon is provided by core, it cannot be
+ *     overridden, but an icon can still be added.
+ *
+ * @return string
+ */
+function icon($icon_name, array $options = array()) {
+  $immutable = !empty($options['immutable']);
+  $icon_path = icon_get_path($icon_name, $immutable);
+
+  if ($icon_path) {
+    unset($options['immutable']);
+    return theme('icon', array(
+      'name' => $icon_name,
+      'path' => $icon_path,
+      'options' => $options,
+    ));
+  }
+
+  return t('Icon %name not found.', array('%name' => $icon_name));
```

Comment:
Am I getting this right, that this text will display on pages?

While it may help to find the source of the problem, I'm not sure about the (public) display. I'm not opposing, just being concerned a bit about visual clutter. :thinking: Or maybe it's OK.

Reply:
Maybe a better solution might be to return an HTML comment so themers have some indication something is wrong without showing to site visitors.